pages/studentAssessmentExam/studentAssessmentExam.js

```diff
@@ -55,6 +126,96 @@ router.post('/', function (req, res, next) {
         res.redirect(res.locals.urlPrefix + '/assessment_instance/' + assessment_instance_id);
       }
     );
+  } else if (req.body.__action === 'join_group') {
```

Comment:
Ditto for this - this looks to be largely copy/pasted from `studentAssessmentHomework.js`. Let's lift this into a shareable function.